nautobot/core/views/generic.py

```diff
@@ -220,6 +220,7 @@ def get(self, request):
         display_filter_params = []
         dynamic_filter_form = None
         filter_form = None
+        hide_hierarchy_ui = "q" in request.GET
```

Comment:
Why does the `q` filter need to be handled separately from the filterset check below?